libFDK/include/FDK_archdef.h

```diff
@@ -107,6 +107,12 @@ amm-info@iis.fraunhofer.de
 #define __arm__
 #endif
 
+#if defined(__AARCH64EL__) && !defined(__arm__)
+#define __TARGET_ARCH_8
+#define __TARGET_ARCH_7E_M
+#define __arm__
+#endif
```

Comment:
Thanks for your pull request - in general it looks really good.

Why do you define `__TARGET_ARCH_7E_M`?

And I feel quite uneasy about defining `__arm__`, since that normally is a compiler built-in define, and one normally doesn't have it defined in aarch64 mode. I understand that having it enabled allows you to just add the new inline assembly snippets in the existing arm directory/headers instead of adding new headers, but since they are distinctly different (and this is using inline assembly and not intrinsics that can be shared), I'm slightly in favor of keeping them separate. (I think that would be favored if I'd try to upstream this patch later as well.)

If you really don't want to create new files, you should rather add the ifdef for `__aarch64__` or similar for all the places where an `arm/*` header is included, but I think new headers would be preferred - see e.g. c15b653b99fbf573ff0aaa2b8c579b0cae693f9c for an example of that.